enterprise/server/iprules/iprules.go

```diff
@@ -421,6 +448,13 @@ func (s *Service) AddRule(ctx context.Context, req *irpb.AddRuleRequest) (*irpb.
 		return nil, err
 	}
 	r.IpRuleId = id
+
+	if sns := s.env.GetServerNotificationService(); sns != nil {
+		if err := sns.Publish(ctx, &snpb.InvalidateIPRulesCache{GroupId: groupID}); err != nil {
+			log.CtxWarningf(ctx, "could not send cache invalidation notification for group %q: %s", groupID, err)
+		}
+	}
```

Comment:
maybe you want to have a method on (s *Service) to do this?

```
(s *Service) publishRuleInvalidation(groupID string) {
	if sns := s.env.GetServerNotificationService(); sns != nil {
		if err := sns.Publish(ctx, &snpb.InvalidateIPRulesCache{GroupId: groupID}); err != nil {
			log.CtxWarningf(ctx, "could not send cache invalidation notification for group %q: %s", groupID, err)
		}
	}
}
```

Reply:
done